### PR TITLE
feat(stdlib): add Range.map

### LINKED ISSUE #616: Stdlib: Implement Range.map
```grain
map : ((Number -> a), Range) -> List<a>
```

### PR DESCRIPTION
**Pull request by Conaclos**
Fix #616

### CODE REVIEW

File: compiler/test/stdlib/range.test.gr [hunk 1]
```diff
@@ -75,3 +75,11 @@ assert results == [2, 3, 4, 5]
 Range.forEach((idx) => {
   fail "Shouldn't be called"
 }, exclusiveSameRange)
+
+let idfn = (a) => a
```
Comment (ospencer) [inline compiler/test/stdlib/range.test.gr:79]: Could we instead do something here that produces a new value? Mapping over `toString` could be a good candidate!

File: stdlib/range.gr [hunk 2]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (ospencer) [inline stdlib/range.gr:1]:
```suggestion
```

**Review by ospencer — CHANGES_REQUESTED** [review]
Very nice, thanks @Conaclos! I've got one request but otherwise looks great 🙌

**Comment by Conaclos** [comment]
Tanks for these suggestions. I updated the code :)

**Review by phated — APPROVED** [review]
Excellent!

**Review by ospencer — APPROVED** [review]
Perfect, thank you!

**Pull request merged**